inspire_json_merger/utils/filterout_utils.py

```diff
@@ -59,24 +64,67 @@ def filter_conflicts_by_path(conflict_list, to_delete_path):
 
 
 def is_to_delete(conflict, keys_path):
-    to_delete = keys_path.split('.')
-    conflict_path = conflict_to_list(conflict)
+    conflict_path = '.'.join(conflict_to_list(conflict))
+    return conflict_path.startswith(keys_path)
```

Comment:
Are you aware that this does not work (if I'm not mistaken) with the example that is given?
For example, `figure.keys` is not matched because `figure.key`, that is the generated `confilct_path` does not start with it.
Moreover, if for whatever reason you'd have something like `figure.records_deleted` it will match `figure.records` too (I guess that has not happened yet, bit it might :{ )

I suggest checking a bit smarter and comparing by sets of the string split by `.` (`['figures', 'keys']`).
about the plurals, probably we can detect when the previous subpath was an integer and add an `s` to the next :/, but that's not really generic (as plurals can be more than just adding an `s`).

Maybe it should not be plural at all, just pass the `to_delete_path` as `figures.key` instead (way less trouble imo).

Reply:
btw. this was there before this change... can be done in another pr (@ammirate maybe has more insight on it)

Reply:
wait, no, it was correct before :/, with a very un-pythonic loop ;)

Reply:
@ammirate, I promised you this:
```python
def is_to_delete(conflict, keys_path):
    conflict_path = conflict_to_list(conflict)
    to_delete = keys_path.split('.')

    for c, d in itertools.zip_longest(conflict_path, to_delete):
        if c is None:
            return True

        if d is None or d != c
            return False

    return True
```

Reply:
More abstractly:
```python
>>> def is_prefix(xs, ys): 
...     for x, y in itertools.izip_longest(xs, ys):
...         if x is None:
...             return True
...         if y is None or y != x:
...             return False
...     return True
... 
>>> is_prefix([], ['foo'])
True
>>> is_prefix(['foo'], ['foo'])
True
>>> is_prefix(['foo'], ['foo', 'bar'])
True
>>> is_prefix(['foo'], [])
False
>>> is_prefix(['foo', 'baz'], ['foo', 'bar'])
False
>>> is_prefix(['foo', 'bar', 'baz'], ['foo', 'bar'])
False
```

Reply:
(Using `six` for Python 3 compatibility is left as an exercise to the reader : )

Reply:
Oh, neat, now that I think about it you can even remove the `y is None` check and just have `if y != x:` !

Reply:
@jacquerie `zip_longest` is what I was looking for but without success, thanks :+1:

Reply:
as @david-caro pointed, that solution doesn't work. I tried with this test and the second assertion fails
```
def test_is_to_delete():
    path = 'figures'
    conflict_list = [
        ('SET_FIELD', ('figures', 0, 'key'), 'figure1.png'),
        ('SET_FIELD', ('figures_attached', 0, 'key'), 'figure2.png')
    ]
    assert is_to_delete(conflict_list[0], path) is True
    assert is_to_delete(conflict_list[1], path) is False
``` 
Could you integrate the solution of @jacquerie?

Reply:
oh, I missed the point, sure, I will modify it.